### PR TITLE
Reference implementation. Closes #8

### LINKED ISSUE #8: Add @Tucker's error method for better exceptions for students

### PR DESCRIPTION
**Pull request by reillysiemens**
### Description

This set of changes includes improvements to error handling by raising the `CalcError` exception with useful error information. It also makes the `test_calculator.py` (formerly `intepreter.py`) tests pass and wires the calculator up to the console script entry point given in `setup.py`. This allows the calculator to be invoked as `calc` after running `pip install -e .`.

### Check List

- [x] All test passed
- [x] Any new functionality or fixes you added is covered by new tests

### CODE REVIEW

**Comment by rawrgulmuffins** [comment]
All tests don't pass? Intentional?

File: calc/__init__.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
 __author__ = 'Reilly Tucker Siemens, Alex LordThorsen'
 __email__ = 'reilly@tuckersiemens.com, alexlordthorsen@gmail.com'
 __version__ = '0.1.0'
+
+from .calc import INTEGER, EOF, PLUS, Calc, CalcError
```
Comment (rawrgulmuffins) [inline calc/__init__.py]: Need to add a Hand Wave TODO to this section for students.

**Comment by rawrgulmuffins** [comment]
Add a couple of small comments. Potentially add me as a push member on your fork so I can do formatting and code tidying when you're not available?

File: calc/__init__.py (hunk 1 above)
Comment (reillysiemens) [inline calc/__init__.py]: Alternatively, what if I just described _why_ we do that right there in the code?

**Comment by reillysiemens** [comment]
Yes, all tests don't pass. The token tests are broken.

**Comment by reillysiemens** [comment]
I'm going to make this a WIP/Don't merge PR instead. I'll change the title again when I feel it's ready.

**Comment by reillysiemens** [comment]
@rawrgulmuffins: What small comments did you have in mind?

**Comment by rawrgulmuffins** [comment]
Reading PR now.

**Comment by rawrgulmuffins** [comment]
This PR was targeted to `Red-Green-Refactor:master`. Needs to be targeted to `red-green-refactor:reference_implementation` rather then`master`.

File: calc/__main__.py [hunk 2]
```diff
@@ -1,5 +1,43 @@
...
+        try:
+            result = calc.parse()
+        except CalcError:
+            traceback.print_exc()  # Show a stack trace on errors.
+        else:
+            print(result)
 
+# Rabbit hole: This pattern below is a Python idiom that basically says, if this
+# module (this file) is being run as the main program (i.e. it was invoked from
+# the command line as python __main__.py, then run the main() function. In
```
Comment (RadicalZephyr) [inline calc/__main__.py]: no ending paren for the begin on the line above

File: calc/calc.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+        Returns:
+            None
+
+        Raises:
+            CalcError: If the character(s) at the given position cannot be
+                tokenized.
+
+        Rabbit hole:
+            This method is prefixed by a ``_`` to denote that it is a "private"
```
Comment (RadicalZephyr) [inline calc/calc.py]: method => method name

or alternatively "The name of this method ..."

File: calc/calc.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+                calculator. Examples include ``INTEGER``, ``PLUS``, or ``EOF``.
+
+        Returns:
+            None
+
+        Raises:
+            CalcError: If ``token_type`` does not match the current token.
+
+        Rabbit hole:
+            This method is prefixed by a ``_`` to denote that it is a "private"
```
Comment (RadicalZephyr) [inline calc/calc.py]: same as above

File: calc/token.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+# error.
+INTEGER, EOF, PLUS = 'INTEGER', 'EOF', 'PLUS'
+
+
+class Token:
+    """
+    A calculator token used to represent the current state of the calculator.
+
+    Note:
+        There is no verification of valid tokens at this point. Later version of
```
Comment (RadicalZephyr) [inline calc/token.py]: versions

**Pull request merged**